saleor/order/tasks.py

```diff
@@ -21,3 +32,58 @@ def send_order_updated(order_ids):
     manager = get_plugins_manager()
     for order in Order.objects.filter(id__in=order_ids):
         manager.order_updated(order)
+
+
+def _batch_ids(iterable, batch_size=1):
+    length = len(iterable)
+    for index in range(0, length, batch_size):
+        yield iterable[index : min(index + batch_size, length)]
+
+
+def _queryset_in_batches(queryset):
+    ids = queryset.values_list("id", flat=True)
+    for ids_batch in _batch_ids(ids, 1000):
+        yield ids_batch
+
+
+def _expire_orders(manager, now):
+    qs = Order.objects.filter(
+        status=OrderStatus.UNCONFIRMED,
+        channel__expire_orders_after__isnull=False,
+        total_charged_amount=Decimal(0),
+    ).select_related("channel")
+    for ids_batch in _queryset_in_batches(qs):
+        with traced_atomic_transaction():
+            orders = Order.objects.filter(id__in=ids_batch).prefetch_related("user")
```

Comment:
Shouldn't be `select_related("user")` instead of `prefetch_related` ?

Reply:
`user` can be null and also idea is to not make those `Order` objects even bigger.
I still wonder if I should do the same with `Channel`